ui/webui/src/components/installation/InstallationLanguage.jsx

```diff
@@ -80,18 +84,28 @@ class LanguageSelector extends React.Component {
     }
 
     updateDefaultSelection () {
-        const languageId = this.props.lang.split("_")[0];
+        const languageId = this.state.lang.split("_")[0];
         const currentLangLocales = this.state.locales.find(langLocales => getLanguageId(langLocales[0]) === languageId);
-        const currentLocale = currentLangLocales.find(locale => getLocaleId(locale) === this.props.lang);
+        const currentLocale = currentLangLocales.find(locale => getLocaleId(locale) === this.state.lang);
 
         this.setState({ selectedItem: getLocaleNativeName(currentLocale) });
     }
 
     render () {
         const { isOpen, languages, locales, selectedItem } = this.state;
         const handleOnSelect = (_, lang) => {
-            this.props.onSelectLang(lang.localeId);
-            this.setState({ selectedItem: lang });
+            /*
+             * When a language is selected from the list, update the backend language,
+             * set the cookie and reload the browser for the new translation file to get loaded.
+             * Since the component will re-mount, we the selectedItem will be set from updateDefaultSelection method.
```

Comment:
I think there is some sort of a typo. :)

Reply:
Fixed.